enhancements/authentication/unauthenticated-subjects.md

```diff
@@ -0,0 +1,123 @@
+
+### Non-Goals
+
+## Proposal
+
+* Background
+  * A user who does not provide enough details for authentication with the kube-apiserver is considered anonymous and will be part of "system:unauthenticated" group.
+  * Both kube-apiserver and openshift-apiserver create a set of cluster roles and cluster role bindings during bootstrap.
+    * This [link](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#default-roles-and-role-bindings) has the list of default cluster roles and cluster role bindings in Kubernetes.
+    * This [link](https://docs.openshift.com/container-platform/4.4/authentication/using-rbac.html#default-roles_using-rbac) has the list of default cluster roles in OpenShift
+  * "rbac.authorization.k8s.io/autoupdate=true" annotation is also added to default policy resources.
+  * On start-up both openshift-apiserver and kube-apiserver reconcile default policy resources. They do the following operations based on the annotation
+    * if the annotation is not modified.
+        * add the missing permissions to cluster roles or add the missing subjects to cluster role bindings.
+    * if the annotation is deleted
+        * add the annotation back if it is missing
+        * add the missing permissions to cluster roles or add the missing subjects to cluster role bindings.
+    * if the annotation is set to "false"
+        * skips reconciliation of resource
+  * reconciliation does not delete any existing permissions or subjects.
+  * Resources with the annotation set to "false" skips above operations. This allows an admin to opt-out of reconciliation for a specific resource.
+  * Detailed behavior can be found in this [link](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#auto-reconciliation).
+
+* Openshift-apiserver operator implements a new RBACController to reconcile following cluster role bindings created by openshift-apiserver.
+  * cluster-status-binding
+  * discovery
+  * system:openshift:discovery
+  * basic-users
+* If the annotation is not modified, the controller removes the unauthenticated subject from the list of subjects and updates the resource in the API server.
+* Customers can opt-out of this reconciliation by setting the annotation to "false".
+
+
+
+### User Stories [optional]
+
+N/A
+
+### Implementation Details/Notes/Constraints [optional]
+
+N/A
+
+### Risks and Mitigations
+
+Because of the earlier releases of OpenShift allowing access to “/” endpoint, some customer implementations might be expecting this privileged access to anonymous users. Removing unauthenticated subject on an upgrade might break some of these implementations. Informing customers beforehand of this change would help the customer to decide better. Customers can either opt-out of reconciliation or change their implementation to match the new requirements. A customer can opt-out of reconciliation by setting the annotation to "false" before the upgrade.
```

Comment:
Maybe list the list of CRBs that the customer might need to change and the CLI invocations required to set the annotation on one or more of those resources? This will aid in development of documentation for the proposed change.

Reply:
Good idea. Add more content based on your comments. Do you think it is better now?